activemodel/lib/active_model/secure_password.rb

```diff
@@ -142,11 +157,30 @@ def has_secure_password(attribute = :password, validations: true)
 
           validates_confirmation_of attribute, allow_blank: true
         end
+
+        # Only generate tokens for records that are capable of doing so (Active Records, not vanilla Active Models)
+        if reset_token && respond_to?(:generates_token_for)
+          generates_token_for :"#{attribute}_reset", expires_in: 15.minutes do
+            public_send(:"#{attribute}_salt")&.last(10)
```

Comment:
@dhh does this mean we'll need `password_digest` and `password_salt` columns?

Reply:
password_digest is already generated in the migration and salt is virtual.